dashboard/app/controllers/programming_classes_controller.rb

```diff
@@ -1,6 +1,7 @@
 class ProgrammingClassesController < ApplicationController
-  before_action :require_levelbuilder_mode_or_test_env
+  before_action :require_levelbuilder_mode_or_test_env, only: [:new, :create, :edit, :update]
   load_and_authorize_resource
+  before_action :set_class_by_keys, only: [:show_by_keys]
```

Comment:
Mostly curious, why did you specify this as a `before_action` instead of just calling it from `show_by_keys`?

Reply:
I think ideally, it should play with `authorize_resource` (might need to rearrange though). But I couldn't get that to work so I'm moving all of it into the controller action for now.

This did make me realize I forgot some tests though...